distribution/lib/Standard/Database/0.0.0-dev/src/Data/Table.enso

```diff
@@ -101,9 +101,20 @@ type Table
     at : Text | Integer -> Column ! No_Such_Column | Index_Out_Of_Bounds
     at self selector=0 = case selector of
         _ : Integer -> self.make_column (self.internal_columns.at selector)
-        _ : Text ->
-            internal_column = self.internal_columns.find if_missing=(Error.throw (No_Such_Column.Error selector)) (p -> p.name == selector)
-            self.make_column internal_column
+        _ -> self.get selector (Error.throw (No_Such_Column.Error selector))
+
+    ## Returns the column with the given name or index.
+
+       Arguments:
+       - selector: The name or index of the column being looked up.
+       - if_missing: The value to use if the selector isn't present.
+    get : Text | Integer -> Any -> Column
+    get self selector=0 ~if_missing=Nothing =
```

Comment:
This is going to be fun with the type signature. `if_missing` can be something else than a `Column` so the return type is not valid really.

But I guess without polymorphic type signatures we can't do much, so let's leave it for now. Just noting that this will bite us in the future - if we get typechecking this signature will definitely be wrong. I guess the correct way would be (not that this is currently supported...)
```suggestion
    get : Text | Integer -> a -> Column | a
    get self selector=0 ~if_missing=Nothing =
```

Reply:
Yeah not a legit syntax yet - I've added `| Any` as a work around for now.